Configuration/Applications/python/ConfigBuilder.py

```diff
@@ -1062,7 +1062,10 @@ def define_Configs(self):
         self.EVTCONTDefaultCFF="Configuration/EventContent/EventContent_cff"
 
         if not self._options.beamspot:
-            self._options.beamspot=VtxSmearedDefaultKey
+            if 'SIM' in self.stepMap or 'GEN' in self.stepMap:
```

Comment:
does `SIM` (alone) needs to have it specified as well?

Reply:
Indeed you are right, this should only be needed for the `GEN` step.

I'm now wondering what will happen to the super-specific case of `GEN:pgen_genonly` introduced by @srimanob in https://github.com/cms-sw/cmssw/pull/39965 where you can run the GEN step without applying any VtxSmearing (it was only used in https://its.cern.ch/jira/browse/PDMVMCPROD-77 AFAIK)...should we somehow take case that into account as well?

Reply:
Both points addressed in c24bd85d2d7e50eeab3fc00978553cf66c0815aa